pallets/asset-index/src/lib.rs

```diff
@@ -681,7 +724,33 @@ pub mod pallet {
             })
         }
 
-        fn remove_asset(_: &T::AssetId) -> DispatchResult {
+        fn remove_asset(
+            asset_id: T::AssetId,
+            units: T::Balance,
+            nav: T::Balance,
+            recipient: Option<T::AccountId>,
+        ) -> DispatchResult {
+            let treasury = Self::treasury_account();
+            ensure!(
+                T::IndexToken::can_slash(&treasury, nav),
+                Error::<T>::InsufficientDeposit
+            );
+
+            if Self::is_liquid_asset(&asset_id) {
+                // Execute the transfer which will take of updating the balance
+                T::RemoteAssetManager::transfer_asset(
+                    recipient.ok_or(Error::<T>::NoRecipient)?,
+                    asset_id,
+                    units,
+                )?;
+            } else {
+                // burn SAFT by withdrawing from the index
+                T::Currency::withdraw(asset_id, &treasury, units)?;
+            }
+
+            // burn index token accordingly, no index token changes in the meantime
+            T::IndexToken::slash(&treasury, nav);
```

Comment:
As a result, treasury slash nav after transferring a liquid asset, should we bind these two operations into one? for example, once we transferring some liquid asset from PINT to other chains, the treasury must slash target nav?

Reply:
Not sure what you mean, but we actually don't have to do any transfers of the liquid assets because the whole xcm pipeline takes care of that.